app/models/chunk.rb

```diff
@@ -10,6 +10,13 @@ class Chunk < ApplicationRecord
   scope :embedded, -> { where.not(vector_id: nil) }
   scope :extracted, ->  { where(entities_extracted: true) }
 
+  # Require embedding content when creating new chunks
+  # Schema doesn't enforce presence for backwards compatibility with existing data
+  validate :explicit_embedding_content?, on: :create
+
+  # Ensure that someone can't change a previously set embedding content back to `nil`
+  attr_readonly :embedding_content
```

Comment:
I was wondering if `excerpt` should be read-only as well, since arguably it is even less valid to change a field that is meant to be a direct excerpt from a document.

Then I started wondering if all these new fields should be read-only - if you can't change the `embedding_content` then you shouldn't be changing the `headers` etc.

Then I wondered if it was worth enforcing this at all, or if we should just trust our code to only update these if it makes sense, like if we're reprocessing a bunch of chunks in place with a new algorithm. :)

Thoughts?

Reply:
Right. And yes, I think it is best if these were read-only as well. While today many will have `nil` values, over time they will only have values when created during ingesttion. Even a re-ingestion will create new chunks.

Reply:
OK. So is the resolution to make all these attributes read-only, or to remove the read-only property from `embedding_content`?

Or did you already make one of those changes and I'm not seeing it in GitHub for some reason?

Reply:
OK... I just looked down.

Minor, but why is `embedding_content` on its own line?

Reply:
Readonly removed as discussed offline.